Kryptonite/Request.swift

```diff
@@ -85,27 +85,25 @@ struct HostAuthVerificationFailed:Error{}
 struct SignRequest:Jsonable {
     var data:SSHMessage //SSH_MSG_USERAUTH_REQUEST
     var fingerprint:String
-    var hostAuth:HostAuth?
+    var verifiedHostAuth:VerifiedHostAuth?
     
     var session:Data
     var user:String
     var digestType:DigestType
     
-    struct InvalidHostAuthSignature:Error{}
+    
+    var isUnknownHost:Bool {
+        return verifiedHostAuth?.hostName == nil
+    }
 
     init(data: Data, fingerprint: String, hostAuth: HostAuth? = nil) throws {
         self.data = SSHMessage(data)
         self.fingerprint = fingerprint
 
         (session, user, digestType) = try SignRequest.parse(requestData: data)
 
-        if let potentialHostAuth = hostAuth{
-            guard try potentialHostAuth.verify(sessionID: session) == true
-            else {
-                throw InvalidHostAuthSignature()
-            }
-
-            self.hostAuth = potentialHostAuth
+        if let potentialHostAuth = hostAuth {
+            self.verifiedHostAuth = try? VerifiedHostAuth(session: session, hostAuth: potentialHostAuth)
```

Comment:
can we add an explanation here that agent-forwarding and _some_ legitimate host key types will show up as unknown host, but we want to phase it out asap?